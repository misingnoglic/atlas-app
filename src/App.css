.App {
  text-align: center;
  background-color: rgb(248 250 252);
}

.App-header {
  min-height: 100vh;
  /* display: flex; */
  /* flex-direction: column;
  align-items: center;
  justify-content: center; */
  font-size: calc(10px + 2vmin);
  /* color: blue */
}

.tabContent {
  display: flex;
  margin-bottom: 60px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  max-width: 600ch;

} 

.footer {
  position: fixed;
  left: 0;
  bottom: 0;
  width: 100%;
  background-color: #C7CEEA;
  color: grey;
  text-align: center;
}

/* unvisited link */
a:link {
  color: #50525E;
}

/* visited link */
a:visited {
  color: #C7CEEA;
}

/* mouse over link */
a:hover {
  color: #B5EAD7;
}

/* selected link */
a:active {
  color: #B5EAD7;
}
