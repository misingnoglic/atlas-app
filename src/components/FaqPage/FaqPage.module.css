.faqPage {
    padding-left: 200px;
    padding-right: 100px;
    padding-top: 10px;
}

.titleText {
    padding-top: 20px ;
    padding-bottom: 10px;
    padding-right: 100px;
    padding-left: 100px;
    text-align: center;
}

/*.subtitleText {*/
/*    !* display: flex; *!*/
/*    padding-top: 10px ;*/
/*    padding-bottom: 10px;*/
/*    padding-right: 10px;*/
/*    padding-left: 10px;*/
/*    text-align: center;*/
/*}*/

.faqText {
    padding: 10px 100px 10px 100px;
    text-align: justify;
    text-justify: inter-word;
}