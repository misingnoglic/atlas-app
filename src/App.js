import React, { useState } from "react";
import PhecodePage from "./components/PhecodePage";
import DeptPage from "./components/DepartmentPage";
import TimePage from "./components/TimePage";
import IbdPage from "./components/IbdPage";
import DeckGlMap from "./components/DeckGlMap";
import About from "./components/About";
import Home from "./components/Home";
import Contact from "./components/Contact";
import AppBar from "@mui/material/AppBar";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import ReactGA from "react-ga";

import "./App.css";

ReactGA.initialize("G-XSD3FL1WE4");

const App = () => {
  const [tabIndex, setTabIndex] = useState(0);

  const handleChangeTabIndex = (event, newIndex) => {
    ReactGA.event({
      category: 'navigation',
      action: 'Switched tab',
      value: tabIndex
    });
    setTabIndex(newIndex);
  };

  const getTabComponent = (tabIndex) => {
    switch (tabIndex) {
      case 0: {
        return <Home />;
      }
      case 1: {
        return <PhecodePage />;
      }
      case 2: {
        return <DeptPage />;
      }
      case 3: {
        return <DeckGlMap />;
      }
      case 4: {
        return <IbdPage />;
      }
      case 5: {
        return <TimePage />;
      }
      case 6: {
        return <About />;
      }

      default: {
        return null;
      }
    }
  };

  const innerComponent = getTabComponent(tabIndex);

  return (
    <div className="App">
      <header className="App-header">
        <AppBar position="static">
          <Tabs
            value={tabIndex}
            onChange={handleChangeTabIndex}
            style={{ backgroundColor: "#C7CEEA", color: "white" }}
          >
            <Tab label="Home" />
            <Tab label="PheCodes" />
            <Tab label="Specialties" />
            <Tab label="Zip Codes" />
            <Tab label="Genetics" />
            <Tab label="Time" />
            <Tab label="FAQ" />
          </Tabs>
        </AppBar>
        <div className="tabContent">{innerComponent}</div>
      </header>
      <div className="footer">
        <p>Â© 2022</p>
      </div>
    </div>
  );
};

export default App;
