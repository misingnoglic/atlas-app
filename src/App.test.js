import renderer from 'react-test-renderer';
import App from "./App";
import FaqPage from "./components/FaqPage";

test("renders learn react link", () => {
  window.ga = jest.fn()
  const tree = renderer
      .create(<App />)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
