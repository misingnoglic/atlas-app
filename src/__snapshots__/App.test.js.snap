// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiTabs-root css-1ujnqem-MuiTabs-root"
        style={
          Object {
            "backgroundColor": "#C7CEEA",
            "color": "white",
          }
        }
      >
        <div
          className="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
          style={
            Object {
              "height": 99,
              "overflow": "scroll",
              "position": "absolute",
              "top": -9999,
              "width": 99,
            }
          }
        />
        <div
          className="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
          onScroll={[Function]}
          style={
            Object {
              "marginBottom": -0,
              "overflow": "hidden",
            }
          }
        >
          <div
            aria-orientation={null}
            className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            onKeyDown={[Function]}
            role="tablist"
          >
            <button
              aria-selected={true}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={0}
              type="button"
            >
              Home
              <span
                className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                style={Object {}}
              />
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              PheCodes
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              Specialties
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              Zip Codes
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              Genetics
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              Time
            </button>
            <button
              aria-selected={false}
              className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex={-1}
              type="button"
            >
              FAQ
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      className="tabContent"
    >
      <div
        className="loadingIndicator"
      >
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </header>
  <div
    className="footer"
  >
    <p>
      Â© 2022 -
       
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="https://github.com/misingnoglic/ibd.la"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Contribute on GitHub
      </a>
    </p>
  </div>
</div>
`;
